src/csl/mod.rs

```diff
@@ -2251,7 +2251,11 @@ pub enum LocatorPayload<'a> {
     Str(&'a str),
     /// An element with the original index of the locator will be yielded. The
     /// consumer can then recognize this and replace it with their own content.
-    Transparent,
+    ///
+    /// Each `Transparent` locator has associated numeric ID. Transparent
+    /// locators with the same ID are considered identical, and with different
+    /// ID are considered different.
+    Transparent(u64),
```

Comment:
If the intention is to use a hash here, we should use the same type used by Typst for hashes.

```suggestion
    Transparent(u128),
```

Reply:
On second thought, let's try this approach so our hands aren't tied (since we already have a lifetime parameter anyway):

```suggestion
    Transparent(&'a dyn Eq),
```

Reply:
Unfortunately, [`Eq` is not dyn compatible](https://doc.rust-lang.org/std/cmp/trait.Eq.html#dyn-compatibility) (think about comparing `&dyn Eq` generated from two different types).

We can introduce generic parameter: `LocatorPayload<'a, T: Eq = ()>`, but I don't think that introducing more generic parameters is the way to go.

We can use `u128` if you want, but I think that we will never need more than `u64`.

Reply:
I would consider a generic parameter with default ‘u64‘  that must implement Copy. Then we know it is reasonably cheap.

Reply:
I've made locator payload type generic (https://github.com/typst/hayagriva/pull/299/commits/11e1095d4b445a03544038b836046d52bba0d17a).

I don't like adding the transparent locator payload type across the whole hayagriva codebase and would rather stick to `u64`. But if you prefer the generic version, you can merge it.

We can additionally add the default type for the generic parameter, but it is not necessary. Typst doesn't need any change to handle the generic parameter — inference does a great job.